src/makeInfinityAdder.js

```diff
@@ -1,10 +1,27 @@
 'use strict';
 
 /***
+ *
  * @return {function}
  */
+
 function makeInfinityAdder() {
-  // write code here
+  let result = 0;
+
+  const adder = (arg) => {
+    if (arg === undefined) {
+      const sum = result;
+      result = 0;
+
+      return sum;
+    }
+
+    result += arg;
+
+    return adder;
+  };
+
+  return adder;
```

Comment:
The `makeInfinityAdder` function correctly creates a closure with the `result` variable and the `adder` function. The `adder` function handles both the accumulation of values and the reset mechanism when called without arguments. This is a good use of closures to maintain state between function calls.